library/unwind/src/lib.rs

```diff
@@ -36,11 +36,34 @@ cfg_if::cfg_if! {
     }
 }
 
-#[cfg(target_env = "musl")]
+#[cfg(all(target_env = "musl", not(feature = "system-llvm-libunwind")))]
 #[link(name = "unwind", kind = "static", cfg(target_feature = "crt-static"))]
+extern "C" {}
+
+#[cfg(all(target_env = "musl", feature = "system-llvm-libunwind"))]
+#[link(name = "unwind", kind = "static-nobundle", cfg(target_feature = "crt-static"))]
+extern "C" {}
+
+#[cfg(all(target_env = "musl", feature = "llvm-libunwind"))]
+#[link(name = "unwind", kind = "static", cfg(not(target_feature = "crt-static")))]
+extern "C" {}
+
+#[cfg(all(
+    target_env = "musl",
+    not(feature = "llvm-libunwind"),
+    not(feature = "system-llvm-libunwind")
+))]
 #[link(name = "gcc_s", cfg(not(target_feature = "crt-static")))]
 extern "C" {}
 
+#[cfg(all(
+    target_env = "musl",
+    not(feature = "llvm-libunwind"),
+    feature = "system-llvm-libunwind"
+))]
+#[link(name = "unwind", cfg(not(target_feature = "crt-static")))]
+extern "C" {}
```

Comment:
What are the possible cases for musl?

| `feature = "llvm-libunwind"` | `feature = "system-llvm-libunwind"` |
| ------------- | ------------- |
| No  | No  |
| No | Yes |
| Yes  | No |
| Yes | Yes |

?

Why are there 5 `extern "C"` blocks for musl then?

Could you write all the `feature = "..."` predicates in `cfg`s explicitly? Otherwise it's hard to understand what happens.

Reply:
Or perhaps it will look better if `cfg_if!` is used instead.